dataplane/pkg/common/egress_interface_test.go

```diff
@@ -9,10 +9,18 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+func TestParseGatewayZero(t *testing.T) {
+	g := NewWithT(t)
+
+	gw, err := parseGatewayIP("0")
+	g.Expect(err.Error()).To(Equal("Failed to parse IP from string"))
+	g.Expect(gw.IsUnspecified()).To(BeTrue())
+}
 func TestParseDefaultGateway(t *testing.T) {
 	g := NewWithT(t)
 
-	gw := parseGatewayIP("010011AC")
+	gw, err := parseGatewayIP("010011AC")
```

Comment:
It's good to add a test case with non-accurate value, as `"010011AS"` or random string

Reply:
👋 @ivanayov thanks for the review, when parsing ```"010011AS"``` it actually parses fine to  ```0.17.0.2``` This is probably not a valid ip, but I am not sure how to deal with this as there is no validity check on ip I can raise a error with any ideas ?

I added a test for string length > 8 and fixed the assert  as it was not asserting the gateway ip string was actually the value expected.

Reply:
@pringlewood yep, actually the error checking is enough